responses/30-makefile-anatomy.md

```diff
@@ -1,82 +1,93 @@
-
-#### Using remakefiles in data science pipelines
+#### Setting up a `targets` data science pipeline
 
 In addition to phases (which we covered in {{ store.structure_activity_url }}), it is important to decompose high-level concepts (or existing scripts) into thoughtful functions and "targets" that form the building blocks of data processing pipelines. A target is a noun we use to describe a tangible output of function, which is often a file or an R object. Targets can be used as an end-product (like a summary map) or as input _into_ another function to create _another_ target. 
 
----
-The simplest version of a `remakefile` (adapted from the [remake repo](https://github.com/richfitz/remake)) might look something like this:
-
-```yaml
-sources:
-  - code.R
-
-targets:
-  all:
-    depends: figure_1.png
-
-  model_RMSEs.csv:
-    command: download_data(out_filepath = "model_RMSEs.csv")
+To setup a `targets` pipeline, you will need to setup the base makefile named `_targets.R` that will declare and orchestrate the rest of the pipeline connections.
 
-  plot_data:
-    command: process_data(in_filepath = "model_RMSEs.csv")
-
-  figure_1.png:
-    command: myplot(out_filepath = "figure_1.png", data = plot_data)
+---
+The simplest version of `_targets.R` might look something like this:
+
+```r
+library(targets)
+source("code.R")
+tar_option_set(packages = c("tidyverse", "stringr", "sbtools", "whisker"))
+
+list(
+  tar_target(
+    model_RMSEs_csv,
+    download_data(out_filepath = "model_RMSEs.csv"),
+    format = "file"
+  ), 
+  tar_target(
+    plot_data,
+    process_data(in_filepath = model_RMSEs_csv),
+  ),
+  tar_target(
+    figure_1_png,
+    make_plot(out_filepath = "figure_1.png", data = plot_data), 
+    format = "file"
+  )
+)
 ```
 
 
-This file defines the relationships between different "targets" (see how the target `model_RMSEs.csv` is an input to the command that creates the target `plot_data`?), tells us where to find any functions that are used to build targets (see that `sources` points you to `code.R`), and isolates the output(s) that _must_ be created in order to complete the `all` target (in this case, it is only the `figure_1.png` file). 
+This file defines the relationships between different "targets" (see how the target `model_RMSEs_csv` is an input to the command that creates the target `plot_data`?), tells us where to find any functions that are used to build targets (see the `source` call that points you to `code.R`), and declares the package dependencies needed to build the different targets (see the `target_option_set` command that passes in a vector of packages). 
```

Comment:
I think I prefer `model_RMSEs_csv` rather than `model_RMSEs.csv` - the later makes it seem like the file name is called `RMSEs.csv` when it might not always be the case (e.g. when file names are long and you don't include the whole file name in the target name)

Reply:
Works for me. There's also the case of `model_RMSEs_csvs`, `modeled_sites_shapefile`, etc. where we'll need to deviate from a simple extension anyway because the target represents multiple files, possibly having multiple extensions. In those cases it also seems nice to have a very clearly symbolic target name. (Not saying these should be introduced in this course, just that our convention ought to be able to accommodate these multi-file targets.)

Reply:
Related to this was the topic of using quotes when initially setting the name of the target. I would personally like to make a plug against that practice since it would mean that sometimes you quote the target name and other times you don't.

Reply:
Ah, good point. I think at the very least I can see a case for using unquoted target names throughout these training modules. Perhaps we introduce the quoting option in ds-pipelines-2, tips and tricks, but then use unquoted everywhere else?
    We (Jake, Sam, and I) had discussed adding quotes to current pipelines because they do make the target name pop out visually a bit better, and it's nice to be able to quickly spot the target names in a makefile, but (1) adding quotes is an imperfect solution, and targets will never be quite as lovely as the remake.yml version in this respect (unless RStudio adds special syntax highlighting for targets someday), and (2) my own efficiency in skimming for unquoted target names has  improved with experience, so maybe this is just a growing pain and we'll all be comfortable with unquoted someday.